bug/kmz-drift-icons-not-up-to-date

edsanville opened the pull request.

edsanville committed:
Update drift icons in exported kmz files

edsanville committed:
Merge remote-tracking branch 'origin/1.y' into bug/kmz-drift-icons-not-up-to-date

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +509,25 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+}
+
+
+export function driftPacketIconStyle(feature: Feature, animatedColor?: string) {
```
Can we add the a comment before this function like the driftspeedtobinindex function  /***/

ferrom-jaia replied:
https://github.com/jaiarobotics/jaiabot/blob/1.y/src/doc/markdown/page_110_coding_standards.md

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -377,18 +379,18 @@ export function getWaypointCircleStyle(feature: Feature<Point>) {
-    const colorBorder = colorNameToHex('black')
+    const colorBorder = '#000'
 
     function getCircleStyle(center: Coordinate, radius: number, color: string, lineWidth: number, addInnerGradientColor: boolean) {
```
Can we add the a comment before this function like the driftspeedtobinindex function  /***/

ferrom-jaia replied:
https://github.com/jaiarobotics/jaiabot/blob/1.y/src/doc/markdown/page_110_coding_standards.md

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -207,8 +209,8 @@ export function hubCommsCircleStyle(feature: Feature<Point>) {
     function getCircleStyle(center: Coordinate, radius: number, color: string, lineWidth: number) {
```
Can we add the a comment before this function like the driftspeedtobinindex function  /***/

ferrom-jaia commented on src/web/jdv/server/kmz.py:
```diff
@@ -164,7 +164,7 @@ def kml_from_task_packets(task_packets: Iterable[TaskPacket]):
 
 
-def write_file(task_packets: Iterable[TaskPacket], output_kmz_path: str):
+def writeTaskPacketsToKMZ(task_packets: Iterable[TaskPacket], output_kmz_path: str):
```
Can we add the a comment before this function like the driftspeedtobinindex function  /***/ (This is python so the equivalent #)

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -37,8 +38,13 @@ async function taskPacketToKMLPlacemarks(taskPacket: LogTaskPacket) {
+    // We omit the file:// here, so that the KMZ can be opened properly in Google Earth
+    const diveIconUrl = 'files/diveIcon.png'
+
+    function getDriftArrowHeadUrl(drift: DriftPacket) {
```
Can we add the a comment before this function like the driftspeedtobinindex function  /***/

edsanville committed:
Added function comments

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -25,6 +26,14 @@ async function getDataUrl(url: string) {
 
+/**
+ * Returns an array of strings, containing the KML code for each feature in a task packet
+ * @date 11/14/2023 - 3:16:59 PM
```
We can remove the date. Github should track this for us.

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -37,8 +46,21 @@ async function taskPacketToKMLPlacemarks(taskPacket: LogTaskPacket) {
+    
+    /**
+     * Returns the path to the appropriate embedded drift arrow icon PNG file
+     * @date 11/14/2023 - 3:14:28 PM
```
We can remove the date. Github should track this for us.

edsanville committed:
Add function description

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -37,8 +46,21 @@ async function taskPacketToKMLPlacemarks(taskPacket: LogTaskPacket) {
+     * Returns the path to the appropriate embedded drift arrow icon PNG file
+     * @date 11/14/2023 - 3:14:28 PM
+     *
+     * @param {DriftPacket} drift drift packet whose speed will be used to find the appropriate icon
```
Duplication of drift* ```drift drift```
Can we add a dash in between the type and the description -

edsanville replied:
Actually the duplication was intentional here, because the first "drift" is the name of the parameter, and the second "drift" is the first word in the description.  It's not necessary though, so I removed it.

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+/**
+ * Returns the drift icon index that should be displayed, given a drift speed.
+ * 
+ * @param driftSpeed Speed of the drift, in m/s
```
needs parameter type {number}
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+ * Returns the drift icon index that should be displayed, given a drift speed.
+ * 
+ * @param driftSpeed Speed of the drift, in m/s
+ * @returns Index into Styles.driftArrowPngs, of the icon sthat should represent this drift
```
needs return type {number}
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+
+/**
+ * Returns an OpenLayers Style for the given Feature
+ * @date 11/14/2023 - 3:35:06 PM
```
We can remove the date. Github should track this for us.

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+ * Returns an OpenLayers Style for the given Feature
+ * @date 11/14/2023 - 3:35:06 PM
+ *
+ * @export
```
We can remove this one @export

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+ * @export
+ * @param {Feature} feature Input drift packet feature
+ * @param {?string} [animatedColor] String indicating what kind of animated image to use.  Set to `black` if you want the animated version of the drift icon.
+ * @returns {*} OpenLayers Style for this drift icon
```
can we change from {*} -> {void}
Can we add a dash in between the type and the description -

edsanville committed:
Added function description

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -25,6 +26,14 @@ async function getDataUrl(url: string) {
+ * Returns an array of strings, containing the KML code for each feature in a task packet
+ * @date 11/14/2023 - 3:16:59 PM
+ *
+ * @async
```
We can remove async.

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -25,6 +26,14 @@ async function getDataUrl(url: string) {
+ * @date 11/14/2023 - 3:16:59 PM
+ *
+ * @async
+ * @param {LogTaskPacket} taskPacket The task packet to process into KML features
```
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -25,6 +26,14 @@ async function getDataUrl(url: string) {
+ *
+ * @async
+ * @param {LogTaskPacket} taskPacket The task packet to process into KML features
+ * @returns {Promise<string[]>} A promise for the array of strings for each KML feature in `taskPacket`
```
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -37,8 +46,21 @@ async function taskPacketToKMLPlacemarks(taskPacket: LogTaskPacket) {
+     * @date 11/14/2023 - 3:14:28 PM
+     *
+     * @param {DriftPacket} drift drift packet whose speed will be used to find the appropriate icon
+     * @returns {string} path to the embedded drift arrow icon file
```
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -175,6 +178,13 @@ export class KMLDocument {
+     * Returns a Blob representing the KML document as a KMZ file
+     *
+     * @async
+     * @returns {Promise<Blob>} A promise for a Blob containing the KMZ file
```
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/jdv/client/src/KMZExport.ts:
```diff
@@ -175,6 +178,13 @@ export class KMLDocument {
+    /**
+     * Returns a Blob representing the KML document as a KMZ file
+     *
+     * @async
```
Can remove async

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+ *
+ * @export
+ * @param {Feature} feature Input drift packet feature
+ * @param {?string} [animatedColor] String indicating what kind of animated image to use.  Set to `black` if you want the animated version of the drift icon.
```
Can we add a dash in between the type and the description -

ferrom-jaia commented on src/web/shared/Styles.ts:
```diff
@@ -507,12 +508,35 @@ export function divePacketIconStyle(feature: Feature, animatedColor?: string) {
+ * @date 11/14/2023 - 3:35:06 PM
+ *
+ * @export
+ * @param {Feature} feature Input drift packet feature
```
Can we add a dash in between the type and the description -

edsanville committed:
Removed @dates from TS function comments

edsanville committed:
Remove redundant "drift"
Convert some python functions to camelCase

edsanville committed:
Added some types to docstrings
Fixed some other docstring issues
Added docstrings and type annotations for JDV API endpoints
Convert API function names to camelCase

edsanville committed:
Added JSDoc strings for all of the shared OpenLayers map styles

ferrom-jaia committed:
Merge branch '1.y' into bug/kmz-drift-icons-not-up-to-date

ferrom-jaia reviewed (approved):
Looks great! Tested on my machine.
